### ISSUE TITLE
Optimal Transport based Loss Function in Texar

### ISSUE THREAD

**Issue by nvidiaman** [asyml/texar#122]
Hello,

Recently people have used optimal transport for sequence to sequence models like in this work : https://openreview.net/pdf?id=S1xtAjR5tX 

How can I implement an optimal transport loss function in texar?

Any way to integrate sequence level loss (not just the word level cross entropy loss) in texar?

Any code I can start with which is close to optimal transport like implementation?

Thank you for your kind help.

**Comment by ZhitingHu** [comment]
Yes. Contributions are very welcome! You may refer to Texar's other [loss functions](https://github.com/asyml/texar/tree/master/texar/losses) to see how an optimal transport loss can be implemented.

Pls checkout the [seq2seq_exposure_bias](https://github.com/asyml/texar/tree/master/examples/seq2seq_exposure_bias) example for implementations of several sequence-level training algorithms
